homeassistant/components/bosch_shc/binary_sensor.py

```diff
@@ -47,3 +71,47 @@ def device_class(self):
             "GENERIC": DEVICE_CLASS_WINDOW,
         }
         return switcher.get(self._device.device_class, DEVICE_CLASS_WINDOW)
+
+
+class BatterySensor(SHCEntity, BinarySensorEntity):
+    """Representation of a SHC battery reporting sensor."""
+
+    @property
+    def unique_id(self):
+        """Return the unique ID of this sensor."""
+        return f"{self._device.serial}_battery"
+
+    @property
+    def name(self):
+        """Return the name of this sensor."""
+        return f"{self._device.name} Battery"
+
+    @property
+    def is_on(self):
+        """Return the state of the sensor."""
+        if (
+            self._device.batterylevel
+            == SHCBatteryDevice.BatteryLevelService.State.NOT_AVAILABLE
+        ):
+            _LOGGER.debug("Battery state of device %s is not available", self.name)
```

Comment:
We don't want side effects like logging in entity state properties. Please move it or remove it.